### PR TITLE
Working on warnings

### LINKED ISSUE #186: make code more readable by adding spaces
We have code like this:

```
      if ( ibmap.eq.0 .OR. ibmap.eq.254 ) then
         allocate(pfld(max(2,ngrdpts)))
         ndpts=0;
         do jj=1,ngrdpts
             intbmap(jj)=0
             if ( bmap(jj) ) then
                intbmap(jj)=1
                ndpts=ndpts+1
                pfld(ndpts)=fld(jj);
             endif
         enddo

```

Spacesareimportantfortheeyetobeabletoeasilyseparatesymbols.It'spossibletoreadtextwithoutspaces,butit'sharderandtakesmoreeffort.Wewantourcodetobereadable.

So we should use spaces in the normal manner.

### LINKED ISSUE #266: compile warnings for test program test_cmplxpack.f90
@BrianCurtis-NOAA can you fix these?

I'm trying to turn on automatic warning checking for this repo, but all existing warnings need to be cleared first..

```
cd /home/ed/NCEPLIBS-g2/b/tests && /usr/bin/gfortran  -I/home/ed/NCEPLIBS-g2/b/src/include_d -I/usr/local/NCEPLIBS-bacio-2.5.0/include -I/usr/local/NCEPLIBS-w3emc-2.9.2/include_d -I/usr/include -g -Wall -fsanitize=address -g -c /home/ed/NCEPLIBS-g2/tests/test_cmplxpack.f90 -o CMakeFiles/test_cmplxpack.dir/test_cmplxpack.f90.o
/home/ed/NCEPLIBS-g2/tests/test_cmplxpack.f90:21:15:

   21 |     fld_orig = (/cmplx(42, 3), cmplx(43, 2), cmplx(44, 1), cmplx(99, 0)/)
      |               1
Warning: Possible change of value in conversion from COMPLEX(4) to REAL(8) at (1) [-Wconversion]
/home/ed/NCEPLIBS-g2/tests/test_cmplxpack.f90:21:17:

   21 |     fld_orig = (/cmplx(42, 3), cmplx(43, 2), cmplx(44, 1), cmplx(99, 0)/)
      |                 1
Warning: Non-zero imaginary part discarded in conversion from ‘COMPLEX(4)’ to ‘REAL(8)’ at (1) [-Wconversion]
/home/ed/NCEPLIBS-g2/tests/test_cmplxpack.f90:21:30:

   21 |     fld_orig = (/cmplx(42, 3), cmplx(43, 2), cmplx(44, 1), cmplx(99, 0)/)
      |                              1
Warning: Non-zero imaginary part discarded in conversion from ‘COMPLEX(4)’ to ‘REAL(8)’ at (1) [-Wconversion]
/home/ed/NCEPLIBS-g2/tests/test_cmplxpack.f90:21:44:

   21 |     fld_orig = (/cmplx(42, 3), cmplx(43, 2), cmplx(44, 1), cmplx(99, 0)/)
      |                                            1
Warning: Non-zero imaginary part discarded in conversion from ‘COMPLEX(4)’ to ‘REAL(8)’ at (1) [-Wconversion]

```

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of  #181 
Part of #132 
Fixes #186 
Fixes #266

Trying to clear up some more warnings...

### CODE REVIEW

**Review by Hang-Lei-NOAA — APPROVED** [review]

File: src/rdieee.F90 [hunk 1]
```diff
@@ -10,49 +10,46 @@
...
-
-     if ( (iexp.gt.0).and.(iexp.lt.255) ) then
-        temp=2.0**(iexp-127)
-        a(j)=sign*temp*(1.0+(two23*real(imant)))
-
-     elseif ( iexp.eq.0 ) then
-        if ( imant.ne.0 ) then
-           a(j)=sign*two126*two23*real(imant)
+  real, parameter :: two23=scale(1.0, -23)
+  real, parameter :: two126=scale(1.0, -126)
```
Comment (BrianCurtis-NOAA) [inline src/rdieee.F90]:
```
  real, parameter :: two23 = scale(1.0, -23)
  real, parameter :: two126 = scale(1.0, -126)
```
spaces before and after = sign

File: src/specpack.F90 [hunk 2]
```diff
@@ -16,95 +16,95 @@
...
 
-  real,intent(in) :: fld(ndpts)
-  integer,intent(in) :: ndpts,JJ,KK,MM
-  integer,intent(inout) :: idrstmpl(*)
-  character(len=1),intent(out) :: cpack(*)
-  integer,intent(out) :: lcpack
+  real, intent(in) :: fld(ndpts)
+  integer, intent(in) :: ndpts, JJ, KK, MM
+  integer, intent(inout) :: idrstmpl(*)
+  character(len=1), intent(out) :: cpack(*)
```
Comment (BrianCurtis-NOAA) [inline src/specpack.F90]: `len = 1` to stay consistent in space before and after = sign

File: src/specunpack.F90 [hunk 3]
```diff
@@ -18,77 +18,73 @@
...
-     !   Calculate Laplacian scaling factors for each possible wave number.
-     !
-     allocate(pscale(JJ+MM))
-     tscale=real(idrstmpl(5))*1E-6
-     do n=Js,JJ+MM
-        pscale(n)=real(n*(n+1))**(-tscale)
+  if (idrstmpl(10) .eq. 1) then           ! unpacked floats are 32-bit IEEE
+     !call g2_gbytesc(cpack, ifld, 0, 32, 0, Ts)
+     call rdieee(cpack, unpk, Ts)          ! read IEEE unpacked floats
+     iofst = 32*Ts
```
Comment (BrianCurtis-NOAA) [inline src/specunpack.F90]: `32 * Ts`

File: src/specunpack.F90 [hunk 4]
```diff
@@ -18,77 +18,73 @@
...
+     call g2_gbytesc(cpack, ifld, iofst, nbits, 0, ndpts-Ts)  ! unpack scaled data
+
+     ! Calculate Laplacian scaling factors for each possible wave number.
+     allocate(pscale(JJ + MM))
+     tscale = real(idrstmpl(5))*1E-6
```
Comment (BrianCurtis-NOAA) [inline src/specunpack.F90]: `tscale = real(idrstmpl(5)) * 1E-6`

File: src/specunpack.F90 [hunk 5]
```diff
@@ -18,77 +18,73 @@
...
+     do n = Js, JJ + MM
+        pscale(n) = real(n*(n + 1))**(-tscale)
```
Comment (BrianCurtis-NOAA) [inline src/specunpack.F90]: `pscale(n) = real(n * (n + 1)) ** (-tscale)`

**Review by BrianCurtis-NOAA — APPROVED** [review]
None of my suggested changes should block merge.

File: src/specunpack.F90 (hunk 5 above)
Comment (edwardhartnett) [reply src/specunpack.F90]: Fixed.

File: src/specunpack.F90 (hunk 4 above)
Comment (edwardhartnett) [reply src/specunpack.F90]: fixed.

File: src/specunpack.F90 (hunk 3 above)
Comment (edwardhartnett) [reply src/specunpack.F90]: Fixed.

File: src/specpack.F90 (hunk 2 above)
Comment (edwardhartnett) [reply src/specpack.F90]: Fixed.

File: src/rdieee.F90 (hunk 1 above)
Comment (edwardhartnett) [reply src/rdieee.F90]: Fixed.

**Pull request merged**